hack/lib/build.sh

```diff
@@ -76,10 +77,25 @@ build_binaries() {
     fi
 
     local target_bin_dir=$(get_binary_dir_with_arch ${YURT_BIN_DIR})
+    local -a statics=()
+    local -a tests=()
+    for binary in "${targets[@]}"; do
+      if [[ "${binary}" =~ "test" ]]; then
+        tests+=(${binary})
+      else
+        statics+=(${binary})
+      fi
+    done
     mkdir -p ${target_bin_dir}
     cd ${target_bin_dir}
-    for binary in "${targets[@]}"; do
+    for binary in "${statics[@]:+${statics[@]}}"; do
       echo "Building ${binary}"
       go build -o $(get_output_name $binary) -ldflags "${goldflags:-}" -gcflags "${gcflags:-}" ${goflags} $YURT_ROOT/${binary}
     done
+    for binary in "${tests[@]:+${tests[@]}}"; do
+      echo "Building ${binary}"
+      local testpkg="$(dirname ${binary})"
+      local filename="$(basename ${binary})"
+      go test -c  -gcflags "${gcflags:-}" ${goflags} -o $filename "$YURT_ROOT/${testpkg}"
+    done
```

Comment:
could you create a function just for test? say `run_test()` and make the build logic looks like following:
`make e2e` -> `hack/make-rules/test.sh` -> `hack/lib/test.sh run_test()`

Reply:
thanks for you comment. I will do like this.